### PR TITLE
docs(contributing): update requirements for contributing

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

Introduces "Requirements" section to our contributing guide for community pull requests. We ❤️  community contributions, but want to make sure they have the proper details up front, before a community member invests their time. 

- Makes `ionic info` section required in the issue template
- Explicitly notes that issues are required for features/bug fixes in the PR template
- Updates Slack link to Discord

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -83,10 +84,16 @@ Without a reliable code reproduction, it is unlikely we will be able to resolve
 
 ## Creating a Pull Request
 
-* We appreciate you taking the time to contribute! Before submitting a pull request, we ask that you please [create an issue](#creating-an-issue) that explains the bug or feature request and let us know that you plan on creating a pull request for it. If an issue already exists, please comment on that issue letting us know you would like to submit a pull request for it. This helps us to keep track of the pull request and make sure there isn't duplicated effort.
+Before creating a pull request, please read our requirements that explains the minimal details to have your PR considered and merged into the codebase. 
 
-* Looking for an issue to fix? Make sure to look through our issues with the [help wanted](https://github.com/ionic-team/ionic/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) label!
+### Requirements
+1. PRs must reference an existing issue that describes the issue or feature being submitted.
+2. PRs must have a reproduction app or example to verify changes against.
```
Comment (amandaesmith3) [inline .github/CONTRIBUTING.md:91]: Ideally, the issue linked to will have a repro already; I assume they don't need to worry about linking it in both places?

Comment (sean-perkins) [reply .github/CONTRIBUTING.md:91]: Correct, should I mention that the linked issue should contain a reproduction app or they must include one?

Comment (amandaesmith3) [reply .github/CONTRIBUTING.md:91]: Yeah, something like that. We just need a repro _somewhere._

**Review by amandaesmith3 — APPROVED** [review]
Thanks for updating this 👍

**Review by liamdebeasi — APPROVED** [review]

**Pull request merged**